### ISSUE TITLE
Events passed to enhance are not correctly destroyed

### ISSUE THREAD

**Issue by gbouteiller** [ciscoheat/sveltekit-superforms#466]
- [✓] Before posting an issue, read the [FAQ](https://superforms.rocks/faq) and search the previous issues.

**Description**
When `events` are added to `use:enhance`, if the form element is created and destroyed multiple times (for example, the form is in a `sheet` or `dialog` or, even simpler, in an `if` like in the **MRE** below), the events are pushed multiple times to `formEvents` but never destroyed.

**MRE**
In that [example](https://www.sveltelab.dev/yhnjbaj4vbn8hcv?files=.%2Fsrc%2Froutes%2F%2Bpage.svelte), if you toggle the form multiple times, you can see that `onUpdated` is called multiple times

**Possible solution**
Instead of returning kitEnhance here :
https://github.com/ciscoheat/sveltekit-superforms/blob/1b7a1585c2661034d71d6f1d6ed39c3d992f5eee/src/lib/client/superForm.ts#L1605
it could be something like this :
```ts
const enhance = kitEnhance("...");
return {
  destroy: () => {
    // same as in onDestroy()
    for (const events of Object.values(formEvents)) {
      events.length = 0;
    }
    // call native enhance destroy
    enhance.destroy();
  }
}
```

What are your thoughts?
By the way, thank you again for this awesome library.

**Comment by ciscoheat** [comment]
Thank you, will take a look and hopefully it will be included in the next release. Glad you like Superforms. :)

**Comment by ciscoheat** [comment]
I've added a fix for the next release. Do you have any good reason not to put the event(s) directly in the options when calling superForm? I'm considering deprecating this "add events in enhance" feature, as I have never found a really good use case for it.

**Comment by ciscoheat** [comment]
I'm not sure I understand still, since `onUpdated` looks like a simple toast notification (maybe you've simplified it), what prevents you from adding it directly?

```ts
let sfUpsert = superForm(data.svUpsert, {
  validators: zodClient(projectUpsertSchema),
  onUpdated: () => toast.success("...")
})
```

**Comment by gbouteiller** [comment]
As you said , I simplified it 🙂. In fact, I use it for multiple interactions on the generic page (closing the sheet, updating some data, etc...). But even if it was a simple toast, I want the toast to be on the generic page as I don't want, if I have 10 collections, to define the same behavior 10 times (hence the generic crud component 🙂).

**Comment by ciscoheat** [comment]
Right, but if you use it in multiple locations, it will append the events to the others as well, and there are internal state that may cause problems too. That's why I'm not sure it's the best to keep. It's was a very early feature.
